### ISSUE TITLE
Make unicorn/consistent-destructuring configurable

### ISSUE THREAD

**Issue by geoffswift** [sindresorhus/eslint-plugin-unicorn#2469]
unicorn/consistent-destructuring seems to trigger in multiple scenarios.  We would ideally be able to control which specific cases are active.

For example this seems to always be good
```js
function example(o) {
  const { a } = o;
  console.log(a);
  // Please do complain that I've already got a const for a 👍 
  console.log(o.a); 
}
```

This scenario in particular I don't like, and would want to turn off:
```js
class Example {
  example() {
    const { a } = this; // rule wants me to write const { a, expensive } = this
    if (a) {
        // this.expensive is not being consistently destructured
        // but i don't want to call the expensive getter unnecessarily in the const above
        console.log(this.expensive); 
    }
  }
    
  get expensive() {
    // Something that requires significant compute
  }
}
```
